File: js/16953.js
const fs = require('fs');

const [A, B] = fs
  .readFileSync('example.txt')
  .toString()
  .trim()
  .split(' ')
  .map(Number);

console.log(bfs());

function bfs() {
  const q = [[1, A]];
  while (q.length !== 0) {
    const [cnt, value] = q.shift();

    if (value === B) {
      return cnt;
    }

    for (let next of [value * 2, value * 10 + 1]) {
      if (next <= B) {
        q.push([cnt + 1, next]);
      }
    }
  }

  return -1;
}

File: c++/16953.cpp
#include <iostream>
#include <queue>

using namespace std;

long long A, B;

long long bfs() {
  queue<pair<long long, long long>> q;
  q.push({1, A});

  while (!q.empty()) {
    long long cnt = q.front().first;
    long long value = q.front().second;
    q.pop();

    if (value == B) {
      return cnt;
    }

    if (value * 2 <= B) {
      q.push({cnt + 1, value * 2});
    }
    if (value * 10 + 1 <= B) {
      q.push({cnt + 1, value * 10 + 1});
    }
  }

  return -1;
}

int main() {
  freopen("example.txt", "r", stdin);

  cin >> A >> B;

  cout << bfs();

  return 0;
}